src/plugins/unified_field_list/public/hooks/use_query_subscriber.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useEffect, useState } from 'react';
+import type { DataPublicPluginStart } from '@kbn/data-plugin/public';
+import type { AggregateQuery, Query, Filter } from '@kbn/es-query';
+
+/**
+ * Hook params
+ */
+export interface QuerySubscriberParams {
+  data: DataPublicPluginStart;
+}
+
+/**
+ * Result from the hook
+ */
+export interface QuerySubscriberResult {
+  query: Query | AggregateQuery | undefined;
+  filters: Filter[] | undefined;
+}
+
+/**
+ * Memorizes current query and filters
+ * @param data
+ */
+export const useQuerySubscriber = ({ data }: QuerySubscriberParams) => {
```

Comment:
Could be a good case for the `useObservable` hook to simplify:
```ts
const { query, filters } = useObservable(data.query.state$);
return { query, filters } as QuerySubscriberResult;
```

Reply:
@davismcphee Thanks for your suggestion. I was going to extend this hook with date range updates in a following PR.